components/supervisor/pkg/ports/ports.go

```diff
@@ -191,10 +191,21 @@ func (pm *Manager) updateState(exposed []ExposedPort, served []ServedPort, confi
 	if exposed != nil && !reflect.DeepEqual(pm.exposed, exposed) {
 		pm.exposed = exposed
 	}
-	if served != nil && !reflect.DeepEqual(pm.served, served) {
-		pm.served = served
-		pm.updateProxies()
+
+	if served != nil {
+		newServed := make(map[uint32]ServedPort)
+		for _, port := range served {
+			current, exists := newServed[port.Port]
+			if !exists || (!port.BoundToLocalhost && current.BoundToLocalhost) {
```

Comment:
nit
```suggestion
if exists {
	continue
}
if !port.BoundToLocalhost && current.BoundToLocalhost {
	continue
}
```

Reply:
it will have other semantics, assignment won't ever happen